worker/caasupgrader/upgrader_test.go

```diff
@@ -121,6 +121,23 @@ func (s *UpgraderSuite) TestUpgraderDowngradePatch(c *gc.C) {
 	s.operatorUpgrader.CheckCall(c, 0, "Upgrade", "machine-0", s.upgraderClient.desired)
 }
 
+func (s *UpgraderSuite) TestUpgraderDowngradeMinor(c *gc.C) {
```

Comment:
So we have tests for downgrade patch and minor.
What about major?
We should allow that but only if >=2.

So we may need to retain a form of the code that was deleted?

Reply:
Yeah, that makes sense - I'll update it.